packages/audiofileplayer/pubspec.yaml

```diff
@@ -5,7 +5,7 @@ homepage: https://github.com/google/flutter.plugins/tree/master/packages/audiofi
 
 environment:
   sdk: '>=2.12.0 <3.0.0'
-  flutter: ">=1.10.0"
+  flutter: ">=3.0.0"
```

Comment:
This means that this plugin would no longer work with previous versions of Flutter. I'm assuming that your change is still valid on Flutter 1 and 2.

Reply:
I have done the same edits as gilescm did but without changing the flutter version and it works ok.

Reply:
Yeah, I don't think there's a way around it. Apologies I should've left a commment.

I found without this change anyone using <3.0.0 can't use the plugin anyway still need the `!` operator would still be needed on the nullable version of `WidgetsBinding.instance`. This way they find out earlier during dependency resolving rather than later.

I saw the same behaviour in [this PR for secure_application](https://github.com/neckaros/secure_application/pull/34). 

I guess its not ideal for backwards compatibility but anyone using <3.0.0 but the previous versions of the package still work and hopefully it wouldn't be too difficult for anyone to update to >=3.0.0 since its been out for awhile now